libraries/botbuilder-dialogs-adaptive/src/actions/throwException.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    ValueExpression,
+    ValueExpressionConverter,
+} from 'adaptive-expressions';
+import {
+    Converter,
+    ConverterFactory,
+    Dialog,
+    DialogConfiguration,
+    DialogContext,
+    DialogEvents,
+    DialogTurnResult,
+} from 'botbuilder-dialogs';
+
+export interface ThrowExceptionConfiguration extends DialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
+    errorValue?: unknown | ValueExpression;
+}
+
+/**
+ * Action which throws an exception declaratively.
+ */
+export class ThrowException<O extends object = {}> extends Dialog<O> implements ThrowExceptionConfiguration {
+    public static $kind = 'Microsoft.ThrowException';
+
+    /**
+     * Initializes a new instance of the [ThrowException](xref:botbuilder-dialogs-adaptive.ThrowException) class.
+     * @param errorValue Optional. Memory property path to use to get the error value to throw.
+     */
+    public constructor(errorValue: unknown) {
+        super();
+        if (errorValue) {
+            this.errorValue = new ValueExpression(errorValue);
+        }
+    }
+
+    /**
+     * Gets or sets an optional expression which if is true will disable this action.
+     */
+    public disabled?: BoolExpression;
+
+    /**
+     * Gets or sets the memory property path to use to get the error value to throw.
+     */
+    public errorValue?: ValueExpression;
+
+    getConverter(property: keyof ThrowExceptionConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'errorValue':
+                return new ValueExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Starts a new [Dialog](xref:botbuilder-dialogs.Dialog) and pushes it onto the dialog stack.
+     *
+     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param options Optional. Initial information to pass to the dialog.
+     * @returns A `Promise` representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: O): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled.getValue(dc.state)) {
+            return await dc.endDialog();
+        }
+
+        let value: unknown;
```

Comment:
Do we want to completely omit the `Error` constructor argument if `value` or `this.errorValue` is undefined?

Reply:
Yes, we can just throw a raw Error in that case.